### PR TITLE
Updated #1385, 2-Collection_Containers.adoc file updated for Primitive sets

### PR DESCRIPTION
**Pull request by sandip1805**
Provide example for creation with different set of available methods. 
Examples provided for different set of operations available in primitive sets.
 
This pull request is against open [issue](https://github.com/eclipse/eclipse-collections/issues/1385)

#1385

### CODE REVIEW

File: docs/2-Collection_Containers.adoc [hunk 1]
```diff
@@ -965,6 +965,64 @@ They are named *{IntHashSet}*, *{FloatHashSet}* etc.
 *{BooleanHashSet}* is implemented using a single integer to hold one of four possible states: ([], [F], [T], or [T, F]).
 All other sets use open addressing and quadratic probing.
 
+There are different ways you can create *IntHashSet*, which are listed as below.
+You can create an primitive sets using **newSetWith** method. You can initialize an *IntHashSet*  using constructor argument. There are 5 different overloaded constructor are available.
+Below examples are provided with *IntHashSet* but all are same for *FloatHashSet* and *BooleanHashSet* as well.
+[source,java]
+----
+// Initialize IntHashSet using newSetWith method, which is taking int vargs.
+IntHashSet intHashSet1 = IntHashSet.newSetWith(1, 2, 3, 4, 5);
```
Comment (donraab) [inline docs/2-Collection_Containers.adoc:974]: Change the types on the left to the interfaces `MutableIntSet` , `MutableFloatSet` and `MutableBooleanSet`.

File: docs/2-Collection_Containers.adoc [hunk 2]
```diff
@@ -965,6 +965,64 @@ They are named *{IntHashSet}*, *{FloatHashSet}* etc.
...
+FloatHashSet floatHashSet1 = FloatHashSet.newSetWith(12.5f, 13.5f, 14.88f, 15.6f);
+BooleanHashSet booleanHashSet = BooleanHashSet.newSetWith(true, false);
+
+// Empty Constructor will initialize IntHashSet with 16 size
+IntHashSet intHashSet2 = new IntHashSet();
```
Comment (donraab) [inline docs/2-Collection_Containers.adoc:979]: Change the type on the left to the interface `MutableIntSet`.

File: docs/2-Collection_Containers.adoc [hunk 3]
```diff
@@ -965,6 +965,64 @@ They are named *{IntHashSet}*, *{FloatHashSet}* etc.
...
+
+// Initialize Constructor with initialCapacity
+IntHashSet intHashSet3 = new IntHashSet(32);
```
Comment (donraab) [inline docs/2-Collection_Containers.adoc:982]: Change the type on the left to the interface `MutableIntSet`.

File: docs/2-Collection_Containers.adoc [hunk 4]
```diff
@@ -965,6 +965,64 @@ They are named *{IntHashSet}*, *{FloatHashSet}* etc.
...
+
+// Initialize Constructor with set of integers using vargs.
+IntHashSet intHashSet4 = new IntHashSet(1, 2, 3, 4, 5, 6);
```
Comment (donraab) [inline docs/2-Collection_Containers.adoc:985]: Change the type on the left to the interface `MutableIntSet`.

File: docs/2-Collection_Containers.adoc [hunk 5]
```diff
@@ -965,6 +965,64 @@ They are named *{IntHashSet}*, *{FloatHashSet}* etc.
...
+
+// Initialize Constructor with IntIterable.
+IntIterable intIterable = intHashSet4; // you can get from any list, set which implement this interface
+IntHashSet intHashSet = new IntHashSet(intIterable);
```
Comment (donraab) [inline docs/2-Collection_Containers.adoc:989]: Change the type on the left to the interface `MutableIntSet`.

File: docs/2-Collection_Containers.adoc [hunk 6]
```diff
@@ -965,6 +965,64 @@ They are named *{IntHashSet}*, *{FloatHashSet}* etc.
...
+
+// Initialize Constructor with existing IntHashSet
+IntHashSet intHashSet5 = new IntHashSet(intHashSet2);
```
Comment (donraab) [inline docs/2-Collection_Containers.adoc:992]: Change the type on the left to the interface `MutableIntSet`.

File: docs/2-Collection_Containers.adoc [hunk 7]
```diff
@@ -965,6 +965,64 @@ They are named *{IntHashSet}*, *{FloatHashSet}* etc.
...
+----
+
+Operation which you can perform on primitive hashsets.
+[source,java]
+----
+// Initialize IntHashSet using newSetWith method, which is taking int vargs.
+IntHashSet intHashSet = IntHashSet.newSetWith(1, 2, 3, 4, 5);
```
Comment (donraab) [inline docs/2-Collection_Containers.adoc:999]: Change the type on the left to the interface `MutableIntSet`.

File: docs/2-Collection_Containers.adoc [hunk 8]
```diff
@@ -965,6 +965,64 @@ They are named *{IntHashSet}*, *{FloatHashSet}* etc.
...
+
+// containsAll check for all element in primitive set
+boolean containsAll = intHashSet.containsAll(5, 4);
+
+// We can do a math operation like min, max on intHashSet
+int min = intHashSet.min();
+int max = intHashSet.max();
+
+// We can filter set with select method, which return filtered set, same way you can use reject as well
+IntHashSet moduloby2 = intHashSet.select(x -> 0 == x % 2);
```
Comment (donraab) [inline docs/2-Collection_Containers.adoc:1015]: Change the type on the left to the interface `MutableIntSet`.

**Review by donraab — CHANGES_REQUESTED** [review]
Thank you for the contribution @sandip1805 ! I have a few changes requested inline. It would be better style to keep the types on the left as interfaces, not implementation classes. Also, there is another way to construct primitive types using primitive factory classes. For instance, you could use `MutableIntSet mutableSet = IntSets.mutable.with(1, 2, 3, 4, 5)` or `ImmutableIntSet immutableSet = IntSets.immutable.with(1, 2, 3, 4, 5)`. It would be great if you could additionally show an example of creating sets using the factories. Thanks!

File: docs/2-Collection_Containers.adoc (hunk 2 above)
Comment (sandip1805) [reply docs/2-Collection_Containers.adoc]: Thank you for your suggestion. I have implemented the suggested comment.

File: docs/2-Collection_Containers.adoc (hunk 3 above)
Comment (sandip1805) [reply docs/2-Collection_Containers.adoc]: Thank you for your suggestion. I have implemented the suggested comment.

File: docs/2-Collection_Containers.adoc (hunk 4 above)
Comment (sandip1805) [reply docs/2-Collection_Containers.adoc]: Thank you for your suggestion. I have implemented the suggested comment.

File: docs/2-Collection_Containers.adoc (hunk 8 above)
Comment (sandip1805) [reply docs/2-Collection_Containers.adoc]: Thank you for your suggestion. I have implemented the suggested comment.

File: docs/2-Collection_Containers.adoc (hunk 7 above)
Comment (sandip1805) [reply docs/2-Collection_Containers.adoc]: Thank you for your suggestion. I have implemented the suggested comment.

File: docs/2-Collection_Containers.adoc (hunk 6 above)
Comment (sandip1805) [reply docs/2-Collection_Containers.adoc]: Thank you for your suggestion. I have implemented the suggested comment.

File: docs/2-Collection_Containers.adoc (hunk 5 above)
Comment (sandip1805) [reply docs/2-Collection_Containers.adoc]: Thank you for your suggestion. I have implemented the suggested comment.

File: docs/2-Collection_Containers.adoc (hunk 4 above)
Comment (sandip1805) [reply docs/2-Collection_Containers.adoc]: Thank you for your suggestion. I have implemented the suggested comment.

File: docs/2-Collection_Containers.adoc (hunk 3 above)
Comment (sandip1805) [reply docs/2-Collection_Containers.adoc]: Thank you for your suggestion. I have implemented the suggested comment.

**Comment by sandip1805** [comment]
> Thank you for the contribution @sandip1805 ! I have a few changes requested inline. It would be better style to keep the types on the left as interfaces, not implementation classes. Also, there is another way to construct primitive types using primitive factory classes. For instance, you could use `MutableIntSet mutableSet = IntSets.mutable.with(1, 2, 3, 4, 5)` or `ImmutableIntSet immutableSet = IntSets.immutable.with(1, 2, 3, 4, 5)`. It would be great if you could additionally show an example of creating sets using the factories. Thanks!

Thank you @donraab for your suggestions. I have created separated section for factories method as well now and changed document based on interface instead of implementation classes.

**Review by sandip1805 — COMMENTED** [review]
Done

File: docs/2-Collection_Containers.adoc [hunk 9]
```diff
@@ -965,6 +965,90 @@ They are named *{IntHashSet}*, *{FloatHashSet}* etc.
 *{BooleanHashSet}* is implemented using a single integer to hold one of four possible states: ([], [F], [T], or [T, F]).
 All other sets use open addressing and quadratic probing.
 
+.There are different ways you can create primitive sets, which are listed as below.
+- You can create primitive sets using **newSetWith** method.
+- You can initialize primitive sets using constructor argument. There are 5 different overloaded constructor are available.
```
Comment (donraab) [inline docs/2-Collection_Containers.adoc:970]: "There are 5 different overloaded constructor are available." -> "There are several overloaded constructors available."

File: docs/2-Collection_Containers.adoc [hunk 10]
```diff
@@ -965,6 +965,90 @@ They are named *{IntHashSet}*, *{FloatHashSet}* etc.
 *{BooleanHashSet}* is implemented using a single integer to hold one of four possible states: ([], [F], [T], or [T, F]).
 All other sets use open addressing and quadratic probing.
 
+.There are different ways you can create primitive sets, which are listed as below.
```
Comment (donraab) [inline docs/2-Collection_Containers.adoc:968]: "as below" -> "below"

File: docs/2-Collection_Containers.adoc [hunk 11]
```diff
@@ -965,6 +965,90 @@ They are named *{IntHashSet}*, *{FloatHashSet}* etc.
...
+// containsAll check for all element in primitive set
+boolean containsAll = intHashSet.containsAll(5, 4);
+
+// We can do a math operation like min, max on intHashSet
+int min = intHashSet.min();
+int max = intHashSet.max();
+
+// We can filter set with select method, which return filtered set, same way you can use reject as well
+MutableIntSet moduloby2 = intHashSet.select(x -> 0 == x % 2);
+MutableIntSet notModuloby2 = intHashSet.reject(x -> 0 == x % 2);
```
Comment (donraab) [inline docs/2-Collection_Containers.adoc:1042]: "notModuloby2" -> "odds"

**Review by donraab — CHANGES_REQUESTED** [review]
Thank you for the changes @sandip1805! A few more changes requested inline. It's looking good so I should be able to merge after these changes. Please squash your commits, and I will approve and merge.

File: docs/2-Collection_Containers.adoc (hunk 11 above)
Comment (sandip1805) [reply docs/2-Collection_Containers.adoc]: Thank you for suggestions, I have updated adoc with suggested changes.

File: docs/2-Collection_Containers.adoc [hunk 12]
```diff
@@ -965,6 +965,90 @@ They are named *{IntHashSet}*, *{FloatHashSet}* etc.
...
+
+// containsAll check for all element in primitive set
+boolean containsAll = intHashSet.containsAll(5, 4);
+
+// We can do a math operation like min, max on intHashSet
+int min = intHashSet.min();
+int max = intHashSet.max();
+
+// We can filter set with select method, which return filtered set, same way you can use reject as well
+MutableIntSet moduloby2 = intHashSet.select(x -> 0 == x % 2);
```
[the remark this answers is not in the document]
Comment (sandip1805) [reply docs/2-Collection_Containers.adoc]: Thank you for suggestions, I have updated adoc with suggested changes.

File: docs/2-Collection_Containers.adoc (hunk 10 above)
Comment (sandip1805) [reply docs/2-Collection_Containers.adoc]: Thank you for suggestions, I have updated adoc with suggested changes.

File: docs/2-Collection_Containers.adoc (hunk 9 above)
Comment (sandip1805) [reply docs/2-Collection_Containers.adoc]: Thank you for suggestions, I have updated adoc with suggested changes.

**Comment by sandip1805** [comment]
> Thank you for the changes @sandip1805! A few more changes requested inline. It's looking good so I should be able to merge after these changes. Please squash your commits, and I will approve and merge.

Hi @donraab , I have implemented suggested changes. Can you please review pull request?

**Comment by sandip1805** [comment]
> Thank you for the changes @sandip1805! A few more changes requested inline. It's looking good so I should be able to merge after these changes. Please squash your commits, and I will approve and merge.

Hi @donraab , I have implemented suggested changes. Can you please review pull request?

**Review by donraab — APPROVED** [review]
LGTM. Thanks for the contribution @sandip1805 !

**Comment by sandip1805** [comment]
> LGTM. Thanks for the contribution @sandip1805 !

Thank you @donraab for your support in this Pull Request. This was my first PR and I am happy that it got merged. Thank you again.

**Pull request merged**